### PR TITLE
Dev oauth wallet

### PR DESCRIPTION
**Pull request by paulgoleary**
Add OAuth enabled flag and event to base access group to manage state of oauth config.

### CODE REVIEW

File: base_access_wallet.sol [hunk 1]
```diff
@@ -178,8 +179,6 @@ contract BaseAccessWalletFactory is Ownable {
     bytes32 public version ="BsAccWltFactory20190506154200ML"; //class name (max 16), date YYYYMMDD, time HHMMSS and Developer initials XX
 
     function createAccessWallet() public returns (address) {
-        return  (new BaseAccessWallet(msg.sender));
+        return (new BaseAccessWallet(msg.sender));
```
Comment (elv-preethi) [inline base_access_wallet.sol]: should this be `return(new ExternalUserWallet(msg.sender))`?

File: base_access_wallet.sol [hunk 2]
```diff
@@ -178,8 +179,6 @@ contract BaseAccessWalletFactory is Ownable {
     bytes32 public version ="BsAccWltFactory20190506154200ML"; //class name (max 16), date YYYYMMDD, time HHMMSS and Developer initials XX
```
Comment (elv-preethi) [inline base_access_wallet.sol]: Is it necessary to update version of baseAccessWalletFactory?

File: base_access_wallet.sol (hunk 1 above)
Comment (paulgoleary) [reply base_access_wallet.sol]: No - the 'external' wallet is more for future work.

File: base_access_wallet.sol (hunk 2 above)
Comment (paulgoleary) [reply base_access_wallet.sol]: Actually both version strings should be updated. Thanks.

Comment (paulgoleary) [reply base_access_wallet.sol]: Wait, no. The version string for the `BaseAccessControlGroup` should be updated. Just did that.
This PR didn't really end up changing `BaseAccessWallet` or its factory.

File: base_access_wallet.sol (hunk 1 above)
Comment (elv-preethi) [reply base_access_wallet.sol]: oh ok

File: base_access_wallet.sol (hunk 2 above)
Comment (elv-preethi) [reply base_access_wallet.sol]: got it

**Pull request merged**